### ISSUE TITLE
SConstruct on darwin insists on Xcode and ignores command line tools

### ISSUE THREAD

**Issue by timj** [GalSim-developers/GalSim#717]
I am having trouble building Galsim on my Mac because I do not have Xcode installed but I am using the Apple command line tools. The problem is that an attempt is made to determine the Xcode version by invoking `xcodebuild` and this fails. Can the Xcode version determination be deferred until it might be needed? If TMV is not linking against BLAS then the whole test is irrelevant. The code is currently:
```python
p = subprocess.Popen(['xcodebuild','-version'], stdout=subprocess.PIPE)
xcode_version = p.stdout.readlines()[0].split()[1]
print 'XCode version is',xcode_version
if (platform.mac_ver()[0] >= '10.7' and #xcode_version < '5.1' and
    '-latlas' not in tmv_link and ('-lblas' in tmv_link or '-lcblas' in tmv_link)):
    print 'WARNING: The Apple BLAS library has been found not to be thread safe on'
    print '         Mac OS versions 10.7+, even across multiple processes (i.e. not'
    print '         just multiple threads in the same process.)  The symptom is that'
    print '         `scons tests` may hang when running nosetests using multiple'
    print '         processes.'
    if xcode_version < '5.1':
        print '         This seems to have been partially fixed with XCode 5.1, so we'
        print '         recommend upgrading to the latest XCode version.  However, even'
        print '         with 5.1, some systems still seem to have problems.'
        env['BAD_BLAS'] = True
    else:
        print '         This seems to have been partially fixed with XCode 5.1, so there'
        print '         is a good chance you will not have any problems.  But there are'
        print '         still occasional systems that fail when using multithreading with'
        print '         programs or modules that link to the BLAS library (such as GalSim).'
        print '         If you do have problems, the solution is to recompile TMV with'
        print '         the SCons option "WITH_BLAS=false".'
```
but changing it to something like:
```python
if (platform.mac_ver()[0] >= '10.7' and
    '-latlas' not in tmv_link and ('-lblas' in tmv_link or '-lcblas' in tmv_link)):
    p = subprocess.Popen(['xcodebuild','-version'], stdout=subprocess.PIPE)
    xcode_version = p.stdout.readlines()[0].split()[1]
    print 'XCode version is',xcode_version
   if xcode_version < '5.1':
```
would at least defer the breakage to a situation where BLAS is involved. I think the only real fix for this is to treat the clang version as a proxy for Xcode version.

**Comment by rmjarvis** [comment]
Sure.  That would be fine.  

And we could also try/except the xcodebuild call and not output the warning if we can't determine the XCode version.  There's certainly nothing fundamental in there that requires the xcode version to decide something.

**Comment by rmjarvis** [comment]
Done on master.  135fd55  This will be included in the next release, v1.4.

**Comment by timj** [comment]
Thanks.
